sources/web/datalab/static/datalab.js

```diff
@@ -277,6 +277,168 @@ function initializePage(dialog, saveFn) {
   });
 }
 
+// constants for minitoolbar operations
+const CELL_METADATA_COLLAPSED = 'hiddenCell';
+const CELL_METADATA_OUTPUT_COLLAPSED = 'collapsed';
+const COLLAPSE_BUTTON_CLASS = 'fa-eye';
+const UNCOLLAPSE_BUTTON_CLASS = 'fa-eye-slash';
+const COLLAPSE_OUTPUT_BUTTON_CLASS = 'fa-compress';
+const UNCOLLAPSE_OUTPUT_BUTTON_CLASS = 'fa-expand';
+const CLEAR_BUTTON_CLASS = 'fa-minus-square-o';
+const RUN_BUTTON_CLASS = 'fa-play';
+const COLLAPSED_CELL_INNER_HTML = '-- Collapsed cell --';
+
+/**
+ * Collapse entire cell
+ */
+function toggleCollapseCell(cell, hide) {
+  if (!(CELL_METADATA_COLLAPSED in cell.metadata)) {
+    return;
+  }
+  let isHidden = typeof(hide) !== 'undefined' ? !hide : cell.metadata[CELL_METADATA_COLLAPSED];
```

Comment:
It took me a fair amount of time to understand this code because it wasn't clear to me that 'isHidden' was a description of the state prior to the function call.

In part, that was due to the fact that the 'isHidden' variable doesn't really represent that... if you call toggleCollapseCell(cell, true), then 'isHidden' will be false even if the cell was already collapsed.

I think this would be clearer if we got rid of the 'isHidden' variable and instead simply set the 'hide' variable to the negation of the 'cell.metadata[CELL_METADATA_COLLAPSED]' expression if it is undefined.